core/controllers/editor.py

```diff
@@ -701,33 +702,20 @@ def get(self, exploration_id):
                 'Unrecognized output format %s' % output_format)
 
 
-class StateDownloadHandler(EditorHandler):
-    """Downloads a state as a YAML string."""
+class StateYamlHandler(EditorHandler):
+    """Given a representation of a state, converts it to a YAML string."""
 
-    def get(self, exploration_id):
+    def get(self):
         """Handles GET requests."""
-        try:
-            exploration = exp_services.get_exploration_by_id(exploration_id)
-        except:
-            raise self.PageNotFoundException
+        state_dict = json.loads(self.request.get('stringified_state'))
+        width = json.loads(self.request.get('stringified_width'))
```

Comment:
It seemed sensible to stringify everything for consistency. If I don't wrap it, it remains a string.